src/PhpParser/NodeFinder/LocalConstantFinder.php

```diff
@@ -28,21 +28,22 @@ public function match(ClassConstFetch $classConstFetch): ?Const_
             return null;
         }
 
-        $constantClassType = $this->nodeTypeResolver->getType($classConstFetch->class);
-        if (! $constantClassType instanceof TypeWithClassName) {
+        $constatName = $this->nodeNameResolver->getName($classConstFetch->name);
+        if ($constatName === null) {
             return null;
         }
+        $const = $this->findConstantByName($class, $constatName);
```

Comment:
this can be moved down to last statement to improve performance

Reply:
good catch!